packages/components/src/Badge/Badge.module.css

```diff
@@ -0,0 +1,103 @@
+.badge {
+  --badge-duration-timing: (var(--animation-duration-slow) / 2)
+    var(--animation-easing-function-ease-in);
+
+  border-radius: var(--spacing-12);
+  color: var(--color-purple-800);
+  display: inline-block;
+  position: relative;
+  font-family: var(--typography-paragraph-extra-small-font-family);
+  font-size: var(--typography-paragraph-extra-small-font-size);
+  font-weight: var(--typography-paragraph-bold-font-weight);
+  letter-spacing: var(--typography-paragraph-extra-small-letter-spacing);
+  line-height: var(--typography-paragraph-extra-small-line-height);
+  padding: 1px var(--spacing-6);
+  min-width: 8px;
+  text-align: center;
+  background-color: var(--badge-background-color, var(--color-gray-300));
+}
+
+.badge.large {
+  display: inline-flex;
+  justify-content: center;
+  border-radius: var(--spacing-48);
+  font-size: var(--typography-data-medium-font-size);
+  line-height: var(--typography-data-medium-line-height);
+  letter-spacing: var(--typography-data-medium-letter-spacing);
+  padding: 2px 1.875rem;
+  width: 24px;
+}
+
+.default {
+  color: var(--color-purple-800);
+}
+
+.active {
+  background-color: var(--color-blue-500);
+  color: var(--color-white);
+}
+
+.reversed.active {
+  background-color: var(--color-green-300);
+  color: var(--color-purple-800);
+}
+
+.dark {
+  background-color: rgba(var(--color-purple-700-rgb), 0.1);
+  color: var(--color-purple-800);
+}
+
+.dot {
+  background-color: var(--color-green-300);
+  padding: var(--spacing-6);
+  min-width: unset;
+}
+
+.dot.large {
+  background-color: var(--color-green-300);
+  padding: 1.875rem;
+  width: unset;
+}
+
+.reversed.dot {
+  background-color: var(--color-green-300);
+}
+
+.reversed.dark {
+  background-color: var(--color-purple-700);
+  color: var(--color-white);
+}
+
+.reversed {
+  background-color: rgba(var(--color-white-rgb), 0.1);
+  color: var(--color-white);
+}
```

Comment:
This has a lower specificity than the combined classes above. I suggest you move this up, as reading the file isn't reflective of the actual cascade.

Another thing - for the classes that change styles when combined with `.reversed`, you can use nesting to make it a little more readable.